### PR TITLE
66 fix monkey config failure logs

### PR DESCRIPTION
**Pull request by tafseer-khan**

### CODE REVIEW

File: protocols/monkey/fixtures/compile/library.go [hunk 1]
```diff
@@ -92,11 +93,18 @@ func (l libraryContext) directory() error {
 	c.ForceGitDir(root)
 	c.ForceContext(l.ctx.universe.Context())
 
-	moduleReader, err := c.HandleLibrary()
+	output, err := c.HandleLibrary()
```
Comment (samyfodil) [inline protocols/monkey/fixtures/compile/library.go:96]: output -> asset

File: protocols/monkey/fixtures/compile/library.go [hunk 2]
```diff
@@ -92,11 +93,18 @@ func (l libraryContext) directory() error {
...
 	if err != nil {
 		return err
 	}
 
+	moduleReader, err := output.Compress(builders.WASM)
```
Comment (samyfodil) [inline protocols/monkey/fixtures/compile/library.go:101]: compressedAsset

File: protocols/monkey/helpers.go [hunk 3]
```diff
@@ -12,3 +17,24 @@ func ToNumber(in interface{}) int {
 	}
 	return 0
 }
+
+func (m *Monkey) storeLogs(r *os.File, errors ...error) (string, error) {
```
Comment (samyfodil) [inline protocols/monkey/helpers.go:21]: os.File -> io.ReadSeeker

Comment (samyfodil) [reply protocols/monkey/helpers.go:21]: storeLogs -> Split into 
appendErrors(r *os.ReadSeeker, errors ...error)
storeLogs(r *os.ReadSeeker)

File: protocols/monkey/helpers.go [hunk 4]
```diff
@@ -12,3 +17,24 @@ func ToNumber(in interface{}) int {
...
+	if len(errors) > 0 {
+		r.Seek(0, io.SeekEnd)
+		r.WriteString("\nMonkey Errors:\n\n")
```
Comment (samyfodil) [inline protocols/monkey/helpers.go:24]: Monkey protocol errors:

File: protocols/monkey/jobs/types.go [hunk 5]
```diff
@@ -44,25 +44,25 @@ type Op struct {
...
 }
 
-type code struct{ Context }
-type website struct{ Context }
-type library struct{ Context }
-type config struct{ Context }
+type code struct{ *Context }
+type website struct{ *Context }
+type library struct{ *Context }
+type config struct{ *Context }
```
Comment (samyfodil) [inline protocols/monkey/jobs/types.go:50]: type config Context
same for others

Comment (samyfodil) [reply protocols/monkey/jobs/types.go:50]: or drop the &config ...

File: protocols/monkey/type.go [hunk 6]
```diff
@@ -61,3 +64,29 @@ func (s *Service) Dev() bool {
...
+		e = append(e, err)
+	}
+
+	return e
+}
+
+func (e *errorsLog) appendAndLog(format string, args ...any) error {
+	if errString := chidori.Format(logger, log.LevelError, format, args...); len(errString) > 0 {
+		err := errors.New(errString)
+
```
Comment (samyfodil) [inline protocols/monkey/type.go:86]: delete

**Review by samyfodil — CHANGES_REQUESTED** [review]

File: protocols/monkey/type.go (hunk 6 above)
Comment (tafseer-khan) [reply protocols/monkey/type.go:86]: Delete what?

**Pull request merged**